### PR TITLE
Use a custom task provider

### LINKED ISSUE #11: Ctrl-Alt-B runs the app even if the build failed
Ctrl-Alt-B runs the app even if the build failed. 

This occurs at least for Rust projects.

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Fixes #11 #3

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
It currently can't deal with spaces in the path:
```bash
flatpak build-init /home/julian/Projekte/GNOME/banner-viewer (Kopie)/.flatpak/repo org.gnome.design.BannerViewerDevel org.gnome.Sdk org.gnome.Platform 3.36/usr/bin/bash: -c: Zeile 0: Syntaxfehler beim unerwarteten Wort `('
```
Error message is in German, but the point is that the input should be quoted 😉

**Comment by bilelmoussaoui** [comment]
Aha good point, i didn't though of that at all... :p

**Comment by Hofer-Julian** [comment]
Sticking with banner-viewer:
After "build-init" comes "update dependencies".
That reads as:
```bash
flatpak-builder --ccache --force-clean --disable-updates --disable-download --build-only --keep-build-dirs --state-dir=/home/julian/Projekte/GNOME/banner-viewer-2/.flatpak/flatpak-builder --stop-at=banner-viewer /home/julian/Projekte/GNOME/banner-viewer-2/.flatpak/repo /home/julian/Projekte/GNOME/banner-viewer-2/build-aux/org.gnome.design.BannerViewerDevel.jsonEmptying app dir '/home/julian/Projekte/GNOME/banner-viewer-2/.flatpak/repo'
```
That reads to me more like build dependecies. It also fails with (still partly German, but the point is that libhandy does not exits):
```bash
Initializing build dir
Committing stage init to cache
Starting build of org.gnome.design.BannerViewerDevel
========================================================================
Building module libhandy in /home/julian/Projekte/GNOME/banner-viewer-2/.flatpak/flatpak-builder/build/libhandy-1
========================================================================

(flatpak-builder:13916): flatpak-builder-WARNING **: 20:40:42.928: Failed to get current git checksum: In Ordner »/home/julian/Projekte/GNOME/banner-viewer-2/.flatpak/flatpak-builder/git/https_gitlab.gnome.org_GNOME_libhandy.git« (Datei oder Verzeichnis nicht gefunden) konnte nicht gewechselt werden
cp: cannot stat '/home/julian/Projekte/GNOME/banner-viewer-2/.flatpak/flatpak-builder/git/https_gitlab.gnome.org_GNOME_libhandy.git': No such file or directory
Error: module libhandy: Der Kindprozess wurde mit Status 1 beendet
```

File: src/commands.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  if (stopAt) {
+    args.push(`--stop-at=${stopAt}`)
+  }
+  args.push(buildDir)
+  args.push(manifestPath)
+
+  return new Command('flatpak-builder', args, cwd)
+}
+
+export const updateDependencies = (manifestPath: string, buildDir: string, cwd: string, stateDir?: string, stopAt?: string): Command => {
```
Comment (Hofer-Julian) [inline src/commands.ts:99]: Yeah, indeed updateDependencies and buildDependencies are the same.
I assume this is a copy-paste error :)

Comment (Hofer-Julian) [reply src/commands.ts:99]: Fixed with the latest commit

File: src/terminal.ts [hunk 2]
```diff
@@ -81,10 +81,19 @@ export class FlatpakTaskTerminal implements vscode.Pseudoterminal {
...
+      proc = child_process.spawn("flatpak-spawn", ["--host", command.name, ...command.arguments], {
+        cwd: command.cwd,
+        shell: '/usr/bin/bash',
+      })
+    }
+    else {
+      proc = child_process.spawn(command.name, command.arguments, {
+        cwd: command.cwd,
+        shell: '/usr/bin/bash',
+      })
```
Comment (bilelmoussaoui) [inline src/terminal.ts:95]: Agh, probably needs fixing regarding the changes i have made in https://github.com/bilelmoussaoui/flatpak-vscode/pull/16/files#diff-80b106dfebcf0f2ffdc8f0b21bc9faa0dea9f9b051480bc437170a7ebf9160afR24-R30 and i think that's the proper place to handle it

**Comment by bilelmoussaoui** [comment]
Ok I'm going to go ahead and merge this as is, it already provides a ton of improvements compared to what we have

**Pull request merged**